egs2/TEMPLATE/spk1/spk.sh

```diff
@@ -0,0 +1,262 @@
+# Speaker model related
+spk_exp=              # Specify the directory path for spk experiment.
+spk_tag=              # Suffix to the result dir for spk model training.
+spk_config=           # Config for the spk model training.
+spk_args=             # Arguments for spk model training.
+
+# [Task dependent] Set the datadir name created by local/data.sh
+train_set=       # Name of training set.
+valid_set=       # Name of validation set used for monitoring/tuning network training.
+test_sets=       # Names of test sets. Multiple items (e.g., both dev and eval sets) can be specified.
+
+. utils/parse_options.sh
+
+if [ $# -ne 0  ]; then
+    log "${help_message}"
+    log "Error: No positional arguments are required."
+        exit 2
+fi
+
+. ./path.sh
+. ./cmd.sh
+
+# Check feature type
+if [ "${feats_type}" = raw  ]; then
+    data_feats=${dumpdir}/raw
+elif [ "${feats_type}" = raw_copy  ]; then
+    # raw_copy is as same as raw except for skipping the format_wav stage
+    data_feats=${dumpdir}/raw_copy
+elif [ "${feats_type}" = fbank_pitch  ]; then
+    data_feats=${dumpdir}/fbank_pitch
+elif [ "${feats_type}" = fbank  ]; then
+    data_feats=${dumpdir}/fbank
+elif [ "${feats_type}" == extracted  ]; then
+    data_feats=${dumpdir}/extracted
+else
+    log "${help_message}"
+    log "Error: not supported: --feats_type ${feats_type}"
+    exit 2
+fi
+
+# Set tag for naming of model directory
+if [ -z "${spk_tag}" ]; then
+    if [ -n "${spk_config}" ]; then
+        spk_tag="$(basename "${spk_config}" .yaml)_${feats_type}"
+    else
+        spk_tag="train_${feats_type}"
+    fi
+fi
+
+# Set directory used for training commands
+spk_stats_dir="${expdir}/spk_stats_${fs}"
+if [ -z "${spk_exp}"  ]; then
+    spk_exp="${expdir}/spk_${spk_tag}"
+fi
+
+# TODO: add speed perturb
+#if [ -n "${speed_perturb_factors}"  ]; then
+#    spk_stats_dir="${spk_stats_dir}_sp"
+#    spk_exp="${spk_exp}_sp"
+#fi
+
+
+if [ ${stage} -le 1  ] && [ ${stop_stage} -ge 1  ] && ! [[ " ${skip_stages} " =~ [[:space:]]1[[:space:]]  ]]; then
+    log "Stage 1: Data preparation for train and evaluation."
+    # [Task dependent] Need to create data.sh for new corpus
+    local/data.sh ${local_data_opts}
+    log "Stage 1 FIN."
+fi
+
+if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
+    if "${skip_train}"; then
+        if "${eval_valid_set}"; then
+            _dsets="${valid_set} ${test_sets}"
+        else
+            _dsets="${test_sets}"
+        fi
+    else
+        _dsets="${train_set} ${valid_set} ${test_sets}"
+    fi
+
+    if [ "${feats_type}" = raw ]; then
+        log "Stage 2: Format wav.scp: data/ -> ${data_feats}"
+        for dset in ${_dsets}; do
+            #if [ "${dset}" = "${train_set}" ] || [ "${dset}" = "${valid_set}" ]; then
+            #    _suf="/org"
+            #else
+            #    _suf=""
+            #fi
+            _suf=""
+            utils/copy_data_dir.sh --validate_opts --non-print data/"${dset}" "${data_feats}${_suf}/${dset}"
```

Comment:
Maybe it is better to keep the `scripts/audio/format_wav_scp.sh` script like in other recipes. Sometimes the audios in wav.scp is in Kaldi's pipe IO style, and should be formated in this stage to raw audios.

Reply:
Or in some cases we may want to convert the sampling rate. It can also be done by this script.